### ISSUE TITLE
Cannot edit Advanced Web Interface since update to ver 4.2.0

### ISSUE THREAD

**Issue by WSCP-GH** [mastodon/mastodon#27228]
### Steps to reproduce the problem

1. Enable Advanced Web Interface in Preferences, Save, go Back to Mastodon
2. I am presented with 4 columns: Search/Post; Home; Notifications; Getting Started
3. I can change content in 4th column by clicking icons at top of 1st column, that's all, no other customization possible.
...

### Expected behaviour

Should be able to customize Advanced Interface

### Actual behaviour

Cannot modify Advanced Interface

### Detailed description

There is no little "X" to close a column; the hamburger icon does nothing; no way to modify default columns in any way.

### Mastodon instance

techhub.social

### Mastodon version

v4.2.0

### Browser name and version

Safari Version 14.1.2 (14611.3.10.1.7)

### Operating system

macOS 10.14.6

**Comment by chantalcoolsma** [comment]
Advanced web interface doesn’t stick overhere since 4.2.0. Advanced is set to on, but if I open the web interface in Safari I get the default interface. This happens every time when I open a new tab/window and open the web interface.

Workaround is to disable and enable advanced web interface, but it’s temporary.

**Comment by ClearlyClaire** [comment]
@WSCP-GH nothing should have changed with advanced interface customization, you should still be able to pin new columns and move/remove pinned columns (by clicking the “Show settings” icon)
![image](https://github.com/mastodon/mastodon/assets/384364/e5ba61dc-dc9f-4c13-883b-253ebd68b11c)

… and looking at the bottom of the column settings:
![image](https://github.com/mastodon/mastodon/assets/384364/22bb3d9d-a48d-4276-8912-a30fe6beab0e)

> Advanced web interface doesn’t stick overhere since 4.2.0. Advanced is set to on, but if I open the web interface in Safari I get the default interface. This happens every time when I open a new tab/window and open the web interface.
> 
> Workaround is to disable and enable advanced web interface, but it’s temporary.

I think this is unrelated. 4.2.0 moved the advanced web interface to `/deck` and unless you specify this, opening e.g. `/home` will open the single-column interface. The single-column layout should explain this in the rightmost corner:
![image](https://github.com/mastodon/mastodon/assets/384364/e906f652-99a0-4515-8334-af83f9174e1d)

**Comment by chantalcoolsma** [comment]
> I think this is unrelated. 4.2.0 moved the advanced web interface to /deck and unless you specify this, opening e.g. /home will open the single-column interface. 

@ClearlyClaire ah I totally missed that. Thanks. Works!

**Comment by ednl** [comment]
> I think this is unrelated. 4.2.0 moved the advanced web interface to `/deck` and unless you specify this, opening e.g. `/home` will open the single-column interface. The single-column layout should explain this in the rightmost corner:

I changed my bookmarks to include `/deck/` but now the interface reverts to single column mode every time, unless I go to `https://mastodon.social/deck/` or `https://mastodon.social/deck/getting-started`. But I want to go to `https://mastodon.social/deck/home`. See attached recording for how that fails in Safari 17.0 on MacOS 14.0.


https://github.com/mastodon/mastodon/assets/9798500/091b72ad-12a3-4192-bd1d-575cac6d7c53

**Comment by ClearlyClaire** [comment]
Oops, this looks like a recently-introduced bug, thanks you for pointing it out!

**Comment by ednl** [comment]
Sure! Did you already patch it and roll it out? Because now my bookmark works fine again, but the server still reports as `v4.2.1-rc1+pr-27261-dd5f312`.

**Comment by ClearlyClaire** [comment]
No, we did not patch it yet, the issue is a caching issue, where the redirect (valid for logged-out users) is incorrectly cached.

**Comment by ednl** [comment]
Ah right, thanks for the clarification. Yeah, it's on & off (hereby retracting my 'fixed' msg..). Glad that you know what it is.

**Comment by oneiros** [comment]
@WSCP-GH Sorry, but we have not managed to reproduce this problem yet. Are you still experiencing this?

**Comment by WSCP-GH** [comment]
Yes, but I’ve been informed I’m using an obsolete browser that the development team is no longer supporting.
